xprof/timeline.cpp

```diff
@@ -44,6 +37,92 @@ static void add_event_end(struct timeline_dispatch *dispatch, perfetto_uuid_t uu
 
 }
 
+static perfetto_uuid_t get_counter_track_uuuid(struct timeline_dispatch *dispatch, std::unordered_map<hp_t, perfetto_uuid_t> &parent_tracks,
+                                               std::unordered_map<hp_device_t, perfetto_uuid_t> counter_tracks, const std::string track_name,
+                                               std::string hostname, uint64_t process_id, thapi_device_id did) {
+  perfetto_uuid_t hp_dev_uuid = 0;
+  auto r = counter_tracks.insert({{hostname, process_id, did}, hp_dev_uuid});
+  auto &potential_uuid = r.first->second;
+  // Exists
+  if (!r.second)
+    return potential_uuid;
+
+  perfetto_uuid_t hp_uuid = 0;
+  auto r2 = parent_tracks.insert({{hostname, process_id}, hp_uuid});
+  auto &potential_uuid2 = r2.first->second;
+  // Didn't exist
+  if (r.second) {
```

Comment:
```suggestion
  if (r2.second) {
```

(may want to use scope)

Reply:
Thanks!